server/src/main/java/org/opensearch/cluster/service/ClusterManagerTaskThrottler.java

```diff
@@ -76,6 +60,24 @@ public ClusterManagerTaskThrottler(
         tasksThreshold = new ConcurrentHashMap<>(128); // setting initial capacity so each task will land in different segment
     }
 
+    /**
+     * To configure a new task for throttling,
+     * * Register task to cluster service with task key,
+     * * override getClusterManagerThrottlingKey method with above task key in task executor.
+     * * Verify that throttled tasks would be retried from data nodes
+     *
+     * Added retry mechanism in TransportClusterManagerNodeAction, so it would be retried for customer generated tasks.
+     *
+     * If tasks are not getting retried then we can register with false flag, so user won't be able to configure threshold limits for it.
+     */
+    protected String registerThrottlingKey(String throttlingKey, boolean retryableOnDataNode) {
+        if (THROTTLING_TASK_KEYS.containsKey(throttlingKey)) {
+            throw new IllegalArgumentException("There is already a Throttling key registered with same name: " + throttlingKey);
+        }
+        THROTTLING_TASK_KEYS.put(throttlingKey, retryableOnDataNode);
+        return throttlingKey;
```

Comment:
you can append some constant string just so it is different string from submitted value.

Reply:
Ack. Will append 'task' suffix in it.

Reply:
I would prefer using a structure for key and move off `String` to ensure the `ClusterManagerTaskThrottler` create an instance of `ThrottlingKey` which is only possible through registration. The key as `String` looks weak in it's current manifestation
```
public static class ThrottlingKey {
   private String taskKey;
   private boolean retryableOnDataNode;
   
    ThrottlingKey(String taskKey, boolean retryableOnDataNode) {
       this.taskKey = taskKey;
       this.retryableOnDataNode = retryableOnDataNode;
    }

   @Override
   public int hashCode() {
   return Objects.hash(taskKey)
  }
}
```

Reply:
Ack @Bukhtawar. If we create new construct for ThrottlingKey (`ClusterManagerThrottlingKey`), we don't need to add constant in throttling key. We can keep task key and throttling key.